ResearchKit/Common/ORKResultPredicate.h

```diff
@@ -41,6 +41,133 @@ ORK_EXTERN NSString *const ORKResultPredicateTaskIdentifierVariableName ORK_AVAI
 #define ORKIgnoreTimeIntervalValue (ORKIgnoreDoubleValue)
 
 /**
+ The `ORKResultIdentifier` class provides a simple wrapper object containing the `taskIdentifier`,
+ `stepIdentifier` and `resultIdentifier` strings. You use this object specify the question result
+ you are interested in when building result predicates. See `ORKResultPredicate` for more
+ information.
+ */
+ORK_CLASS_AVAILABLE
+@interface ORKResultIdentifier : NSObject <NSSecureCoding, NSCopying>
```

Comment:
I feels `[ORKResultIdentifier  identifierWithResultIdentifier: ]` is a little bit confusing. Two `resultIdentifer` actually have different meanings. 

Rename  `ORKResultIdentifier` to 
`ORKResultPredicateIdentfier`? 
`ORKResultPredicateReference`? 
`ORKResultReference`? 
`ORKResultDescriptor`? 

I am also open to other names as well.